GUIDELINES.md

```diff
@@ -1,3 +1,694 @@
+new List<int>();
+```
+
+**Not:**
+
+```c#
+Initialize( someObject );	
+object[ i ];
+new List< int >();
+```
+
+Do use space to separate type parameters to generic types:
+
+**For example:**
+
+```c#
+var someDictionary = new Dictionary<ObjectId, Object>();
+```
+
+**Not:**
+
+```c#
+var someDictionary = new Dictionary<ObjectId,Object>();
+```
+
+Do use single spaces in expressions liberally:
+
+**For example:**
+
+```c#
+if (a + b > someMethod())
+{
+    // ...
+}
+```
+
+**Not:**
+
+```c#
+if (a+b>someMethod())
+{
+    // ...
+}
+```
+
+## Enums
+
+* Use singular names for enums.
+* Do not explicitly specify a type of an enum or values of enums.
+
+**For example:**
+
+```c#
+public enum Direction
+{
+    North,
+    East,
+    South,
+    West
+}
+```
+
+**Not:**
+
+```c#
+public enum Direction : long
+{
+    North = 1,
+    East = 2,
+    South = 3,
+    West = 4
+}
+```
+
+## *#region* directive
+
+`#region` is a preprocessor directive that lets you specify a block of code that you can expand or collapse when using the outlining feature of the Visual Studio Code Editor. It makes the code easier to organize and navigate by other developers.
+
+### Syntax of *#region*
+
+Follow the Microsoft [standard format](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-region) of region directive.
+
+**For example:**
+
+```c#
+#region Lifecycle methods 
+
+	public async void SomeMethodAsync()
+	{
+	    
+	}
+     
+	public async void AnotherMethodAsync()
+	{
+	    
+	}
+	         
+#endregion 
+```
+
+### Organization of *#region*
+
+- Order of the `#region` should be identical in both interface and class files.
+- **All of the methods** under `#region` should be in **alphabetical order**.
+
+## *using* directive
+
+* `using` directives may or may not be in alphabetical order
+* Always roup using directives by common prefix, with shorter namespaces coming before longer ones
+* Preffered, namespaces should be ordered in increasing order of platform specificity, with .NET namespaces first, followed by library or component namespaces, then Xamarin namespaces, finally application namespaces
+
+**For example:**
+
+```c#
+using System;
+using System.Xml;
+using System.Collections.Generic;
+
+using SomeLib;
+using SomeLib.Extensions;
+
+using MonoTouch.Foundation;
+using MonoTouch.UIKit;
+
+using MyApp;
+```
+
+**Not:**
+
+```c#
+using MyApp;
+using SomeLib.Extensions;
+using MonoTouch.Foundation;
+using System.Collections.Generic;
+using System;
+using System.Xml;
+using MonoTouch.UIKit;
+using SomeLib;
+```
+
+## Nomenclature
+
+For naming, you should follow [Microsoft Naming Guidelines](http://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines).
+
+### Special cases
+
+Despite adopting Microsoft naming conventions as general code development guideline, there some special cases that you are required to some rules in order to be more consistent on the framework’s usability.
```

Comment:
Beyond the Microsoft naming conventions for general code development, there are special rules you should follow to ensure consistency and usability of the framework.